uportal-war/src/main/java/org/jasig/portal/layout/dlm/remoting/ChannelListController.java

```diff
@@ -131,78 +129,124 @@ public void setAuthorizationService(IAuthorizationService authorizationService)
     }
 
     /**
-     * 
-     * @param request
-     * @param type
-     * @return
+     * Original, pre-4.3 version of this API.  Always returns the entire contents
+     * of the Portlet Registry, including uncategorized portlets,  to which the
+     * user has access.  Access is based on the SUBSCRIBE permission.
      */
-    @RequestMapping(method = RequestMethod.GET)
-    public ModelAndView listChannels(WebRequest webRequest, HttpServletRequest request, @RequestParam(value="type",required=false) String type) {
+    @RequestMapping(value="/portletList", method=RequestMethod.GET)
+    public ModelAndView listChannels(WebRequest webRequest, HttpServletRequest request,
+            @RequestParam(value="type",required=false) String type) {
+
         if(type != null && TYPE_MANAGE.equals(type)) {
             throw new UnsupportedOperationException("Moved to PortletRESTController under /api/portlets.json");
         }
-        IPerson user = personManager.getPerson(request);
 
-        Map<String,SortedSet<?>> registry = getRegistry(webRequest, user);
+        final IPerson user = personManager.getPerson(request);
+        final PortletCategory rootCategory = portletCategoryRegistry.getTopLevelPortletCategory();
+        final Map<String,SortedSet<?>> registry = getRegistry(webRequest, user, rootCategory, true);
+
+        // Since type=manage was deprecated channels is always empty but retained for backwards compatibility
+        registry.put("channels", new TreeSet<ChannelBean>());
 
         return new ModelAndView("jsonView", "registry", registry);
     }
 
-    private Map<String,SortedSet<?>> getRegistry(WebRequest request, IPerson user) {
+    /**
+     * Updated version of this API.  Supports an optional 'categoryId' parameter.
+     * If provided, this URL will return the portlet registry beginning with the
+     * specified category, including all descendants, and <em>excluding</em>
+     * uncategorized portlets.  If no 'categoryId' is provided, this method
+     * returns the portlet registry beginning with 'All Categories' (the root)
+     * and <em>including</em> uncategorized portlets.  Access is based on the
+     * SUBSCRIBE permission.
+     *
+     * @since 4.3
+     */
+    @RequestMapping(value="/v4-3/dlm/portletRegistry.json", method=RequestMethod.GET)
+    public ModelAndView getPortletRegistry(WebRequest webRequest, HttpServletRequest request,
+            @RequestParam(value="categoryId", required=false) String categoryId) {
```

Comment:
Would there be an advantage in representing the `categoryId` in the path rather than as a request parameter?

`/v4-3/portletRegistry/{categoryId}/`, perhaps?